packages/node_modules/@webex/internal-plugin-conversation/src/conversation.js

```diff
@@ -1115,30 +1130,65 @@ const Conversation = WebexPlugin.extend({
    * @private
    * @returns {Promise<Array<Conversation>>}
    */
-  _list(options) {
+  async _list(options) {
     options.qs = Object.assign({
       personRefresh: true,
       uuidEntryFormat: true,
       activitiesLimit: 0,
       participantsLimit: 0
     }, options.qs);
 
-    return this.request(options)
-      .then((res) => {
-        if (!res.body || !res.body.items || res.body.items.length === 0) {
-          return [];
-        }
+    const res = await this.request(options);
+
+    if (!res.body || !res.body.items || res.body.items.length === 0) {
+      return [];
+    }
+
+    let list = res.body.items.slice(0);
+
+    if (last(list).published < list[0].published) {
+      list.reverse();
+    }
+
+    if (res.body.additionalUrls) {
+      let limit = 0;
+
+      if (options.limit) {
```

Comment:
Brief, high-level comments on each of these blocks would be nice.

Reply:
Each 'if' statement below? I can do that.